src/main/scala/TestServer.scala

```diff
@@ -27,12 +28,17 @@ object TestServer extends App {
   implicit val executionContext: ExecutionContext = system.executionContext
   implicit val materializer: Materializer = Materializer(classicSystem)
 
-  val (queue, source) = Source.queue[Int](50, OverflowStrategy.fail).preMaterialize()
+  val (queue, source0) =
+    Source.queue[Int](50, OverflowStrategy.fail)
+      .scan(0)(_ + _)
+      .toMat(BroadcastHub.sink)(Keep.both)
+      .run()
+  val source = source0.buffer(1000 /* Tune as needed */, OverflowStrategy.fail)
 
   system.scheduler.scheduleWithFixedDelay(0.seconds, 1.second)(() => queue.offer(1))
   val wsHandler = path("ws") {
-    handleWebSocketMessages(Flow.fromSinkAndSource(Sink.ignore, source.map(_ => TextMessage("Hey!"))))
-  }
+    handleWebSocketMessages(Flow.fromSinkAndSourceCoupled(Sink.ignore, source.map(i => TextMessage(s"[$i] Hey!"))))
```

Comment:
Using `fromSinkAndSourceCoupled` can help with closing the connection sooner when the browser closes the connection. It won't help if you not also decrease the `linger-timeout` as seen above.